### ISSUE TITLE
JSON bodies

### ISSUE THREAD

**Issue by mdtusz** [adamhalasz/diet#8]
I can't seem to accept a properly parsed JSON body in a POST request:

Sending a request with  `Content-Type: application/json` in the header and 
```
{
    "hello": "world"
}
```

I would expect to be able to access this through `$.body.hello` but it seems to be parsed incorrectly with output like this:

`{ '\'hello\': \'world\'': 'undefined' }`

Is this something that will require middleware?

**Comment by adamhalasz** [comment]
Hi,

That is weird. It should work without a middleware. 

Are you encoding the json on client side? 

Could you send me an example request, including the client code that initiates the request and the actual headers it generates?

Thanks,
Adam

P.S Setting the Content-Type to application/json will not affect the parsing.

**Comment by mdtusz** [comment]
Here's my code:

```
var diet = require('diet');
var server = diet();

server.listen(3000);

server.post('/', function ($) {
  $.json($.body);
});
```

Then when I either run `curl http://localhost:3000/ -d '{"hello":"world"}'` or send the request using postman, the result is always ` {"{\"hello\":\"world\"}":"undefined"}% {"{\"hello\":\"world\"}":"undefined"}`.

The code runs fine when POSTing a standard url-encoded form body, but not with raw JSON.

**Comment by adamhalasz** [comment]
Hi,

Problem solved in v0.9.3. 

If the Content-Type is `Content-Type:application/json` then diet will interpret the request body as a JSON content *({"hello":"world"})* and it will try to parse it with `JSON.parse()`.

If the Content-Type is specified as `Content-Type:application/x-www-form-urlencoded` then the request body will be parsed as a URL Encoded String *(hello=world&day=1)* and it will be parsed with `$.qs.parse`. This is the default content-type for html forms and browsers should specify this header when the form is sent.

If **NO** Content-Type is specified `$.body` will contain the `Raw` request body and you are free to parse it however you want.

You're example will work if you specifify the header `Content-Type:application/json` as such in curl:
```
curl http://localhost:3000/ -d '{"hello":"world"}' -H "Content-Type: application/json"
```

and you're result should be:
```
{"hello":"world"}
```

In Postman you need to specify the header too:

![postman diet.js post raw json body](https://cloud.githubusercontent.com/assets/839018/7671301/a10fd36a-fcc9-11e4-96aa-d59c7c29e2ac.png)

Let me know if you have any questions!

Thanks,
Adam

**Comment by mdtusz** [comment]
Thanks, works now!
